codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostMultipartRequestDecoder.java

```diff
@@ -805,6 +795,35 @@ private InterfaceHttpData findMultipartDisposition() {
         }
     }
 
+    private Attribute getContentDispositionAttribute(String... values) {
+        Attribute attribute;
+        String name = cleanString(values[0]);
+        String value = values[1];
+
+        boolean shouldDecode = false;
+        int last = name.length() - 1;
+        if (name.charAt(last) == '*') {
+            shouldDecode = true;
+            name = name.substring(0, last);
+        }
+
+        // See http://www.w3.org/Protocols/rfc2616/rfc2616-sec19.html
+        if (HttpHeaderValues.FILENAME.contentEquals(name)) {
+            // filename value is quoted string so strip them
+            if (!shouldDecode) {
```

Comment:
IIUC the RFC 5987 prescribe decoding not only `filename` header value? Otherwise we should not cut `*` from name..

Reply:
I thought about that too. But I'm not sure that's the case (what other headers are there? they aren't being tested for flow control in this code). I decided to just cover this case and establish how it might be done if someone encountered this problem with other headers.

Reply:
I mean, if we want apply new RFC only for `filename` header, we should not change the `name` attribute for other headers. But now you are doing this: `name = name.substring(0, last);`

Reply:
Ah yes that makes sense. I think I avoid this now. Could add a test for it...